src/poller_base.hpp

```diff
@@ -160,7 +162,7 @@ class poller_base_t
 };
 
 //  Base class for a poller with a single worker thread.
-class worker_poller_base_t : public poller_base_t
+class worker_poller_base_t : public virtual poller_base_t
```

Comment:
This did intentionally not use virtual inheritance, which increases the cost of virtual functions calls. Can you explain why using virtual inheritance has become necessary? Can it be avoided? This applies also to severals other classes that have been changed to use virtual inheritance in this commit.

Reply:
The main issue is that worker_poller_base_t implements poller_base_t and *_poller class implements worker_poller_base_t and the *_base_poller which implements also poller_base_t.
We can avoid that by explicitly override the functions in *_poller. Can be done